### PR TITLE
Share all files in folder on folder check

### LINKED ISSUE #3428: Better course material checkboxes
It is not easy for an instructor to share a lot of files to students if the files are not zipped, as every file needs to be checked individually.

Screenshot from 2019-03-19 20-48-34

It might be nice if there are some kind of tree view style checkboxes, where the value in the upper level overrides all values of the lower level if the checkbox of the upper directory is checked.

image

### PR DESCRIPTION
**Pull request by xprilion**
Closes #3428 

Added checkbox labelled "Share All to student on" beside all folder names.

This allows replicating the functionality as mentioned in the issue.

https://www.useloom.com/share/d1e8e1f3aeca4c039ecf536d4786d3e5

### CODE REVIEW

File: site/app/templates/course/CourseMaterials.twig [hunk 1]
```diff
@@ -131,6 +136,16 @@
         }
     }
 
+    function shareAllToOther(id) {
+        // pass folderid to the function
+        // this will find all elements with the classname same as the id and click their checkboxes
+        // this recursively sets the permissions for all files and folders
+
+        $('.'+id).each(function(){
```
Comment (shwehtom89) [inline site/app/templates/course/CourseMaterials.twig]: This code should keep track of which check marks are marked first. Currently it only changes each child box into the opposite state.

Comment (xprilion) [reply site/app/templates/course/CourseMaterials.twig]: Will push fix soon! :)

File: site/app/templates/course/CourseMaterials.twig [hunk 2]
```diff
@@ -64,6 +64,11 @@
             </a>
             {% if userGroup == 1 %}
                 <a onclick='newDeleteCourseMaterialForm("{{ core.buildUrl({'component': 'misc', 'page': 'delete_course_material_folder', 'dir': 'course_materials', 'file': dir, 'path': folderPath }) }}", "{{ dir }}");'> <i class="fas fa-trash" aria-hidden="true" style="font-size: 16px; margin: 5px;"></i></a>
+                <input id="share_all_checkbox_{{ id }}" class="{{ parentId }}" type="checkbox"
+                        {% if ( (fileShares[path] == "1") )%}
+                            checked="true"
+                        {% endif %}
+                       onclick="shareAllToOther('{{ id }}', '{{ path }}')"/><label for="latest-events">Share All to student on</label>
```
Comment (shwehtom89) [inline site/app/templates/course/CourseMaterials.twig]: If you want have the label "Share All to student on", it implies that the user should be able to specify a date to share the whole folder. 

See code for sharing a single file
```suggestion
onclick="shareToOther('{{ id }}', '{{ path }}')" name="ch1"/><label for="latest-events">Share to student on</label>
                <input name="release_date" id="date_to_release_{{ id }}" class="date_picker" type="text" size="26" value="{{ fileReleaseDates[path]}}" onChange="setNewDateTime(this, '{{ path }}')"/>
```
Notice the above input field. 
Additionally, if you choose to share a whole folder, the release dates of the children files/subfolders should sync up to be at earliest the new specified date of the parent folder.

**Comment by bmcutler** [comment]
Thanks for the contribution, but we will close this PR because the solution is incomplete.  You can re-open the PR when the notes above are addressed.

**Pull request closed without merging**